Discover Flatcar Linux properly for hostname

dongsupark opened the pull request:
##### SUMMARY

To avoid issues with Flatcar Container Linux being unable to be found,
detect Flatcar distro name especially for hostname, just like CoreOS
Container Linux was supported.

Fixes https://github.com/ansible/ansible/issues/69516

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
module

Akasurde reviewed (changes requested):
Could you please add a change log entry for this change? Rest LGTM

dongsupark commented:
@Akasurde  I have added a changelog entry.
Thanks for the review.

Commit:
Discover Flatcar Linux properly for hostname

To avoid issues with Flatcar Container Linux being unable to be found,
detect Flatcar distro name especially for hostname, just like CoreOS
Container Linux was supported.

See also https://github.com/ansible/ansible/issues/69516

dongsupark commented:
Rebased, and adapted its unit test for the new test format.

Akasurde commented:
@dongsupark Thanks for the contribution.
